modyn/trainer_server/internal/grpc/trainer_server_grpc_servicer.py

```diff
@@ -255,11 +257,22 @@ def store_final_model(
                 logger.error(f"Could not store final model from training id {training_id}.")
                 return StoreFinalModelResponse(valid_state=False)
 
+            self._cleanup_stored_models(training_id, final_checkpoint_path)
+
             return StoreFinalModelResponse(valid_state=True, model_id=register_response.model_id)
 
         logger.error(f"Could not find final checkpoint of training with ID {training_id}.")
         return StoreFinalModelResponse(valid_state=False)
 
+    def _cleanup_stored_models(self, training_id: int, final_checkpoint_path: pathlib.Path) -> None:
+        pretrained_model_path = self._training_dict[training_id].pretrained_model_path
+        if pretrained_model_path and pretrained_model_path.is_file():
+            os.remove(pretrained_model_path)
+            logger.info(f"Successfully deleted pretrained model at {pretrained_model_path}")
+        if final_checkpoint_path.is_file():
+            os.remove(final_checkpoint_path)
+            logger.info(f"Successfully deleted final model at {final_checkpoint_path}")
+
```

Comment:
I deleted the pretrained model as well as the final model (if they exist) as they are both persistent on model storage at this point. The checkpoint files are not touched.
Can we go with this?

Reply:
+1